Reusable github action for the *.owasp.yml files to validate metadata

abhayymishraa opened the pull request:
- [x] Implemented the github Action (Tested)
 - [x]  Completed validation function 
  
 Test results 
  [Sample Action Run](https://github.com/abhayymishraa/test-action-www-repo/actions/runs/16805854993)

arkid15r commented on .github/actions/validate_owasp_schema/action.yaml:
use `validate` as a name

arkid15r reviewed (changes requested):
Let's make it a Docker container action:
 -  https://docs.github.com/en/actions/tutorials/use-containerized-services/create-a-docker-container-action
 - https://github.com/actions/hello-world-docker-action

arkid15r reviewed (changes requested):
Please add some tests -- it's hard to do refactoring w/o them.

arkid15r commented on tests/validator_test.py:
Rename it to action_test.py

arkid15r replied:
Actually, I suggest keeping it consistent w/ the current structure -- create `action` directory and 3 files for chapter/committee/project files tests. Both file should include just 2 test cases: positive/negative (see `committee_test.py` or `project_test.py` as an example). The test data should be placed under data/action/<entity> directory too.

abhayymishraa commented:
Hi @arkid15r , instead of renaming tests/validator_test.py, I renamed it to action_test.py and placed it under the action directory. I also removed some cases that were already covered in other test files (chapter_test.py, committee_test.py, and project_test.py  related to validity ), keeping only those specifically related to scenarios where the action needs to work as expected, such as test_validation_fails_when_no_file_found and test_validation_fails_when_multiple_files_found. I’m happy to make any further changes if needed.
